tests/unit/scm/test_scm.py

```diff
@@ -0,0 +1,54 @@
+from dvc.repo.experiments import ExpRefInfo
+from dvc.scm import iter_revs
+
+
+def test_iter_revs(
+    tmp_dir,
+    scm,
+):
+    """
+    new         other
+     │            │
+    master (tag) ────┘
+     │
+    root
+    """
+    tmp_dir.scm_gen("foo", "init", commit="init")
+    rev_root = scm.get_rev()
+    tmp_dir.scm_gen("foo", "old", commit="old")
+    rev_old = scm.get_rev()
+    scm.checkout("new", create_new=True)
+    tmp_dir.scm_gen("foo", "new", commit="new")
+    rev_new = scm.get_rev()
+    scm.checkout("master")
+    scm.tag("tag")
+    scm.checkout("other", create_new=True)
+    tmp_dir.scm_gen("foo", "other", commit="new")
+    rev_other = scm.get_rev()
+
+    ref = ExpRefInfo(rev_root, "exp1")
+    scm.set_ref(str(ref), rev_new)
+    ref = ExpRefInfo(rev_root, "exp2")
+    scm.set_ref(str(ref), rev_old)
+
+    gen = iter_revs(scm, [rev_root, "new"], 1)
+    assert gen == {rev_root: rev_root, rev_new: "new"}
+    gen = iter_revs(scm, ["new"], 2)
+    assert gen == {rev_new: "new", rev_old: rev_old}
+    gen = iter_revs(scm, ["other"], -1)
+    assert gen == {rev_other: "other", rev_old: rev_old, rev_root: rev_root}
+    gen = iter_revs(scm, ["tag"])
+    assert gen == {rev_old: "tag"}
+    gen = iter_revs(scm, all_branches=True)
+    assert gen == {rev_old: "master", rev_new: "new", rev_other: "other"}
+    gen = iter_revs(scm, all_tags=True)
+    assert gen == {rev_old: "tag"}
+    gen = iter_revs(scm, all_commits=True)
+    assert gen == {
+        rev_old: rev_old,
+        rev_new: rev_new,
+        rev_other: rev_other,
+        rev_root: rev_root,
+    }
+    gen = iter_revs(scm, all_experiments=True)
+    assert gen == {rev_new: rev_new, rev_old: rev_old, rev_root: rev_root}
```

Comment:
https://github.com/iterative/dvc/blob/79ebbc24e563c7ed3c918f9c3545b355e795794d/dvc/repo/experiments/utils.py#L127-L128

Not understand, why the baseline also been included in `all_experiments`?

Reply:
It's because `scm.branch_revs(branch, parent)` is not inclusive. It walks backwards and includes any revs up until it reaches `parent`